pkg/reposync/conditions.go

```diff
@@ -123,16 +123,28 @@ func setCondition(rs *v1beta1.RepoSync, condType v1beta1.RepoSyncConditionType,
 	if condition.Status != status {
 		condition.Status = status
 		condition.LastTransitionTime = lastUpdate
-		conditionTransited = true
+		condition.Reason = reason
+		condition.Message = message
+		condition.Commit = commit
+		condition.ErrorSourceRefs = errorSources
+		condition.ErrorSummary = errorSummary
+		condition.LastUpdateTime = lastUpdate
+		transitioned = true
+		updated = true
+	} else {
+		copy := condition.DeepCopy()
+		copy.Reason = reason
```

Comment:
Add unit test case for update on each of these fields to verify the equality comparison. Ditto for `rootsync`

Reply:
added

Reply:
I understand so I won't push too hard on this, but it would be nice to have more coverage here (e.g. to ensure that the equality comparisons work as expected and aren't broken in subsequent changes)

Reply:
I added SetSyncing tests. They weren't being tested at all. :stuck_out_tongue: 

Feel free to add some expanded tests with the suggest feature or an added commit. I was initially using DeepEqual to avoid needing to test every field...

Reply:
Thank you for adding the unit tests, more coverage is better :)

I think even if we were using DeepEqual then we should still unit test this to verify each field is getting correctly set and compared.